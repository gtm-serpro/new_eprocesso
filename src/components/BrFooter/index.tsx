import Icon from "@/common/Icon";
import { useMtProps } from "@/hooks/useMtProps";
import { useSpreadProps } from "@/hooks/useSpreadProps";
import IMtProps from "@/types/IMtProps";
import { ColSize } from "@/types/layout";
import classNames from "classnames";
import React from "react";
import Container from "../../common/Container";
import CustomTag from "../../common/CustomTag";
import Col from "../../common/Grid/Col";
import Row from "../../common/Grid/Row";
import Divider from "../BrDivider";
import Item from "../BrItem";
import List from "../BrList";

export interface BrFooterProps extends React.HTMLAttributes<HTMLDivElement>, IMtProps {
  /** URL da Logo */
  urlLogo?: string;
  /** Links do footer */
  links?: Array<{
    category: string;
    size?: ColSize;
    items: Array<{
      label: string;
      link: string;
    }>;
  }>;
  /** Redes sociais do footer. */
  socialNetworks?: Array<{
    icon: string;
    link: string;
    name: string;
  }>;
  /** Texto ao fim do footer, sobre a licen√ßa ou termos de uso. */
  userLicenseText?: React.ReactNode;
  /** Imagens no fim do footer. */
  footerImages?: Array<{
    url: string;
    link: string;
    name: string;
  }>;
  /** Tema invertido */
  inverted?: boolean;
}

const BrFooter = React.forwardRef<HTMLDivElement, BrFooterProps>(
  (
    {
      className,
      children,
      urlLogo,
      links,
      socialNetworks,
      userLicenseText,
      footerImages,
      inverted = false,
      ...props
    },
    ref
  ) => {
    const mtProps = useMtProps(props);
    const spreadProps = useSpreadProps(props);

    return (
      <footer
        ref={ref}
        className={classNames("br-footer", { inverted: inverted }, className, ...mtProps)}
        {...spreadProps}
      >
        <Container fluid="lg">
          <div className="logo">{urlLogo && <img src={urlLogo} alt="Logo" />}</div>
          {links && (
            <List horizontal>
              {links.map((link, index) => (
                <Col key={index} md={link.size || 2}>
                  <Item link="#">
                    <div className="content text-down-01 text-bold text-uppercase">
                      {link.category}
                    </div>
                    <div className="support">
                      <i className="fas fa-angle-down" aria-hidden="true"></i>
                    </div>
                  </Item>
                  <List>
                    <Divider className="d-md-none" />
                    {link.items.map((item, index) => (
                      <Item key={index} link={item.link}>
                        {item.label}
                      </Item>
                    ))}
                  </List>
                </Col>
              ))}
            </List>
          )}
          <div className="d-none d-sm-block">
            <Row className="align-items-end justify-content-between" padding={{ t: 5, b: 5 }}>
              <Col className="social-network">
                {socialNetworks && (
                  <p className="text-up-01 text-extra-bold text-uppercase">Redes Sociais</p>
                )}
                {socialNetworks?.map((socialNetwork, index) => (
                  <a
                    key={index}
                    href={socialNetwork.link}
                    className="mr-3"
                    aria-label={`Rede Social ${socialNetwork.name}`}
                  >
                    <Icon icon={socialNetwork.icon} size="2x" />
                  </a>
                ))}
              </Col>
              {footerImages && (
                <Col className="assigns text-right">
                  {footerImages?.map((footerImage, index) => (
                    <CustomTag
                      key={index}
                      tagName={footerImage.link ? "a" : "div"}
                      href={footerImage.link}
                    >
                      <img className="ml-4" src={footerImage.url} alt={footerImage.name} />
                    </CustomTag>
                  ))}
                </Col>
              )}
            </Row>
          </div>
        </Container>
        {userLicenseText && (
          <>
            <Divider my={3} />
            <Container fluid="lg">
              <div className="info">
                <div className="text-down-01 text-medium pb-3">{userLicenseText}</div>
              </div>
            </Container>
          </>
        )}
        {children}
      </footer>
    );
  }
);

export default BrFooter;
