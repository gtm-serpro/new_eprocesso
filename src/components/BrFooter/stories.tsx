import { Meta, Story } from "@storybook/react";
import BrFooter, { BrFooterProps } from ".";

export default {
  title: "Layout e Tipografia/Footer",
  component: BrFooter,
} as Meta;

const Template: Story<BrFooterProps> = (args) => <BrFooter {...args} />;

export const Padrao = Template.bind({});
Padrao.storyName = "Padrão";
Padrao.args = {
  urlLogo: "https://cdngovbr-ds.estaleiro.serpro.gov.br/design-system/images/logo-negative.png",
  links: [
    {
      category: "Categoria 1",
      items: [
        {
          label: "Ad deserunt nostrud",
          link: "#",
        },
        {
          label: "Nulla occaecat eiusmod",
          link: "#",
        },
        {
          label: "Nulla occaecat eiusmod",
          link: "#",
        },
      ],
    },
    {
      category: "Categoria 2",
      items: [
        {
          label: "Ex qui laborum consectetur aute commodo",
          link: "#",
        },
        {
          label: "Nulla occaecat eiusmod",
          link: "#",
        },
        {
          label: "Ex qui laborum consectetur aute commodo",
          link: "#",
        },
      ],
    },
    {
      category: "Categoria 3",
      items: [
        {
          label: "Ad deserunt nostrud",
          link: "#",
        },
        {
          label: "Ex qui laborum consectetur aute commodo",
          link: "#",
        },
        {
          label: "Nulla occaecat eiusmod",
          link: "#",
        },
      ],
    },
    {
      category: "Categoria 4",
      items: [
        {
          label: "Ad deserunt nostrud",
          link: "#",
        },
        {
          label: "Ex qui laborum consectetur aute commodo",
          link: "#",
        },
        {
          label: "Nulla occaecat eiusmod",
          link: "#",
        },
      ],
    },
    {
      category: "Categoria 5",
      items: [
        {
          label: "Ex qui laborum consectetur aute commodo",
          link: "#",
        },
        {
          label: "Nulla occaecat eiusmod",
          link: "#",
        },
        {
          label: "Nulla occaecat eiusmod",
          link: "#",
        },
      ],
    },
    {
      category: "Categoria 6",
      items: [
        {
          label: "Ex qui laborum consectetur aute commodo",
          link: "#",
        },
        {
          label: "Nulla occaecat eiusmod",
          link: "#",
        },
        {
          label: "Nulla occaecat eiusmod",
          link: "#",
        },
      ],
    },
  ],
  socialNetworks: [
    { name: "Facebook", link: "#", icon: "fab fa-facebook-square" },
    { name: "Twitter", link: "#", icon: "fab fa-twitter-square" },
    { name: "Linkedin", link: "#", icon: "fab fa-linkedin" },
  ],
  footerImages: [
    {
      link: "#",
      name: "Footer Image 1",
      url: "https://cdngovbr-ds.estaleiro.serpro.gov.br/design-system/images/logo-assign-negative.png",
    },
    {
      link: "#",
      name: "Footer Image 2",
      url: "https://cdngovbr-ds.estaleiro.serpro.gov.br/design-system/images/logo-assign-negative.png",
    },
  ],
  userLicenseText: (
    <>
      Texto destinado a exibição de informações relacionadas à&nbsp;<b>licença de uso.</b>
    </>
  ),
};
