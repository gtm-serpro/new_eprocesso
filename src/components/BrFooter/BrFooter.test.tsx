import { render } from "@testing-library/react";

import BrFooter from ".";

describe("<BrFooter />", () => {
  it("deveria renderizar o footer padrão", () => {
    const { container } = render(
      <BrFooter
        urlLogo="/some-image"
        links={[
          {
            category: "Categoria 1",
            items: [
              {
                label: "Ad deserunt nostrud",
                link: "#",
              },
              {
                label: "Nulla occaecat eiusmod",
                link: "#",
              },
              {
                label: "Nulla occaecat eiusmod",
                link: "#",
              },
            ],
          },
        ]}
        socialNetworks={[
          { name: "Facebook", link: "#", icon: "fab fa-facebook-square" },
          { name: "Twitter", link: "#", icon: "fab fa-twitter-square" },
          { name: "Linkedin", link: "#", icon: "fab fa-linkedin" },
        ]}
        userLicenseText={
          <>
            Texto destinado a exibição de informações relacionadas à&nbsp;<b>licença de uso.</b>
          </>
        }
        footerImages={[{ name: "Image 1", link: "#", url: "/some-image" }]}
      />
    );
    expect(container.firstChild).toHaveClass("br-footer");
    expect(container.firstChild).toMatchSnapshot();
  });
});
