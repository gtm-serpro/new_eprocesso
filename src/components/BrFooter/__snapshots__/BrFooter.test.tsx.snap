// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrFooter /> deveria renderizar o footer padrão 1`] = `
<footer
  class="br-footer"
>
  <div
    class="container-lg"
  >
    <div
      class="logo"
    >
      <img
        alt="Logo"
        src="/some-image"
      />
    </div>
    <div
      class="br-list horizontal"
      id="list_____1"
      role="list"
    >
      <div
        class="col-md-2"
      >
        <a
          class="br-item"
          href="#"
          role="listItem"
        >
          <div
            class="content"
          >
            <div
              class="flex-fill"
            >
              <div
                class="content text-down-01 text-bold text-uppercase"
              >
                Categoria 1
              </div>
              <div
                class="support"
              >
                <i
                  aria-hidden="true"
                  class="fas fa-angle-down"
                />
              </div>
            </div>
          </div>
        </a>
        <div
          class="br-list"
          id="list_____2"
          role="list"
        >
          <div
            class="br-divider d-md-none"
          />
          <a
            class="br-item"
            href="#"
            role="listItem"
          >
            <div
              class="content"
            >
              <div
                class="flex-fill"
              >
                Ad deserunt nostrud
              </div>
            </div>
          </a>
          <a
            class="br-item"
            href="#"
            role="listItem"
          >
            <div
              class="content"
            >
              <div
                class="flex-fill"
              >
                Nulla occaecat eiusmod
              </div>
            </div>
          </a>
          <a
            class="br-item"
            href="#"
            role="listItem"
          >
            <div
              class="content"
            >
              <div
                class="flex-fill"
              >
                Nulla occaecat eiusmod
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="d-none d-sm-block"
    >
      <div
        class="row align-items-end justify-content-between pt-5 pb-5"
      >
        <div
          class="social-network col"
        >
          <p
            class="text-up-01 text-extra-bold text-uppercase"
          >
            Redes Sociais
          </p>
          <a
            aria-label="Rede Social Facebook"
            class="mr-3"
            href="#"
          >
            <i
              aria-hidden="true"
              class="fa-2x fab fa-facebook-square"
            />
          </a>
          <a
            aria-label="Rede Social Twitter"
            class="mr-3"
            href="#"
          >
            <i
              aria-hidden="true"
              class="fa-2x fab fa-twitter-square"
            />
          </a>
          <a
            aria-label="Rede Social Linkedin"
            class="mr-3"
            href="#"
          >
            <i
              aria-hidden="true"
              class="fa-2x fab fa-linkedin"
            />
          </a>
        </div>
        <div
          class="assigns text-right col"
        >
          <a
            href="#"
          >
            <img
              alt="Image 1"
              class="ml-4"
              src="/some-image"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    class="br-divider my-3"
  />
  <div
    class="container-lg"
  >
    <div
      class="info"
    >
      <div
        class="text-down-01 text-medium pb-3"
      >
        Texto destinado a exibição de informações relacionadas à 
        <b>
          licença de uso.
        </b>
      </div>
    </div>
  </div>
</footer>
`;
